tests/integration_tests.rs

```diff
@@ -0,0 +1,785 @@
+use std::collections::BTreeMap;
+
+use seccompiler::SeccompCmpArgLen::*;
+use seccompiler::SeccompCmpOp::*;
+use seccompiler::{
+    apply_filter, sock_filter, BpfProgram, Error, SeccompAction, SeccompCondition as Cond,
+    SeccompFilter, SeccompRule,
+};
+use std::convert::TryInto;
+use std::env::consts::ARCH;
+use std::thread;
+
+// The type of the `req` parameter is different for the `musl` library. This will enable
+// successful build for other non-musl libraries.
+#[cfg(target_env = "musl")]
+type IoctlRequest = i32;
+#[cfg(not(target_env = "musl"))]
+type IoctlRequest = u64;
+
+// We use KVM_GET_PIT2 as the second parameter for ioctl syscalls in some unit tests
+// because has non-0 MSB and LSB.
+const KVM_GET_PIT2: u64 = 0x8070_ae9f;
+const KVM_GET_PIT2_MSB: u64 = 0x0000_ae9f;
+const KVM_GET_PIT2_LSB: u64 = 0x8070_0000;
+
+const FAILURE_CODE: i32 = 1000;
+
+const EXTRA_SYSCALLS: [i64; 6] = [
+    libc::SYS_rt_sigprocmask,
+    libc::SYS_sigaltstack,
+    libc::SYS_munmap,
+    libc::SYS_exit,
+    libc::SYS_rt_sigreturn,
+    libc::SYS_futex,
+];
+
+fn validate_seccomp_filter(
+    rules: Vec<(i64, Vec<SeccompRule>)>,
+    validation_fn: fn(),
+    should_fail: Option<bool>,
```

Comment:
nit: It is a bit hard to understand how to use this parameter, specifically what is the meaning of `None`. We could define an enum in which we have 3 options (with some better names maybe):

```rust
enum Return {
    Errno(u32),
    NoErrno,
    NA,
}
```